blocks/rich-text/index.js

```diff
@@ -541,6 +569,11 @@ export class RichText extends Component {
 				if ( event.shiftKey || ! this.props.onSplit ) {
 					this.editor.execCommand( 'InsertLineBreak', false, event );
 				} else {
+					// For type writing offect, save the root node offset so it
+					// can the position can be scrolled to in the next focussed
+					// instance.
+					offsetTop = rootNode.getBoundingClientRect().top;
```

Comment:
While this works at the moment, nothing guarantee that it will work in the future, imagine if the `onSplit` callback of the RichText component inserts something else that a `RichText`, say a `PlainText` or another unrelated block, how this would work? would it be a problem?

I don't have an alternative implementation at the moment but it seems fragile to rely on a global variable like this and assume a certain behavior in parent blocks.